### ISSUE TITLE
Change volume action doesn't work on OnePlus/Oppo/Realme

### ISSUE THREAD

**Issue by micku7zu** [micku7zu/QuickCursor#12]
# Issue description

The "Change volume" action doesn't work outside of Quick Cursor app on OnePlus/Oppo/Realme device.
This issue happens only on Oxygen OS, Color OS and Realme UI (all 3 are based on the same system).

**This is currently being investigated**. I don't have more details on it yet.

**Comment by rohitvarma1986** [comment]
Hi, any updates on this ? 
Thanks

**Comment by micku7zu** [comment]
No update, this is probably a bug or a feature of this OS. I can't do anything from Quick Cursor to change this. The OS controls this.

**Comment by nisargsuthar** [comment]
Interesting find, for me it works when the quickcursor app is in view (like splitscreen) as well as when the app is open. 

I've also tried the same after giving it background permission with same results.

**Comment by micku7zu** [comment]
Yes, based on other feedback I've got from other Oneplus/Oppo/Realme users, it seems that this is indeed a feature of the OS. They implemented specifically this way, only the foreground app can change the volume of the device. 

An app which is not in foreground (so a background app or an app which draws over other apps like QC) can't change the volume of the device. 

I can't bypass this restriction as a 3rd party app (not a system app). The only option is to raise this with Oneplus/Oppo/Realme. They could add a special permission or something, but nothing that I can do on Quick Cursor. 

Sorry!

**Comment by nisargsuthar** [comment]
The dev of precisevolume also faced this issue and has implemented a workaround. Interesting read here: https://precisevolume.phascinate.com/blog/update-2-0-0-beta-4

**Comment by nisargsuthar** [comment]
> When you get the update, please test it and provide feedback.

Works flawlessly, thanks for the quick fix!

**Comment by micku7zu** [comment]
Can you please check your Volume action settings. Go to edit it and see if the 'Oneplus background fix' is enabled or not? 

Also, please try to save it with it disabled then save it with it enabled again, just to be sure that it is actually saved with it enabled. 

Thanks!
